src/main/java/com/cburch/logisim/soc/nios2/Nios2State.java

```diff
@@ -180,48 +175,56 @@ public int getIenable() {
     }
 
     public void setIenable(int value) {
-      if (ienable == value) return;
-      ienable = value;
-      repaint();
+      if (ienable != value) {
+        ienable = value;
+        repaint();
+      }
     }
 
     public int getIpending() {
       return ipending;
     }
 
     public void setIpending(int value) {
-      if (ipending == value) return;
-      ipending = value;
-      repaint();
+      if (ipending != value) {
+        ipending = value;
+        repaint();
+      }
     }
 
     public int getControlRegister(int index) {
-      switch (index) {
-        case 0 : return status;
-        case 1 : return estatus;
-        case 2 : return bstatus;
-        case 3 : return ienable;
-        case 4 : return ipending;
-        default: return 0;
-      }
+      return switch (index) {
+        case 0 -> status;
+        case 1 -> estatus;
+        case 2 -> bstatus;
+        case 3 -> ienable;
+        case 4 -> ipending;
+        default -> 0;
+      };
     }
 
     public void setControlRegister(int index, int value) {
       switch (index) {
-        case 0 : setStatus(value);
-                 break;
-        case 1 : estatus = value;
-                 break;
-        case 2 : bstatus = value;
-                 break;
-        case 3 : ienable = value;
-                 break;
+        case 0:
+          setStatus(value);
+          break;
+        case 1:
+          estatus = value;
+          break;
+        case 2:
+          bstatus = value;
+          break;
+        case 3:
+          ienable = value;
+          break;
+        default:
+          // FIXME: what to do by default? throw Exception?
```

Comment:
Yes, I think this makes sense, although it never should happen

Reply:
Added.